lib/maths/CBoostedTreeImpl.cc

```diff
@@ -362,12 +364,15 @@ std::size_t CBoostedTreeImpl::estimateMemoryUsage(std::size_t numberRows,
 }
 
 std::size_t CBoostedTreeImpl::correctedMemoryUsage(double memoryUsageBytes) {
-    // We compute the correction coefficient as a sigmoid function: ca. 1.0 until
-    // 10mb, ca 16.0 after 1000mb to this end we need to shift and scale using the
-    // magic numbers below.
-    double correctionCoefficient{
-        CTools::logisticFunction(memoryUsageBytes / (1024 * 1024), 100, 550) * 15 + 1};
-    return static_cast<std::size_t>(memoryUsageBytes / correctionCoefficient);
+    // We use a piecewise linear function of the estimated memory usage to compute
+    // the corrected value. The values are selected in a way to reduce over-estimation
+    // on small jobs to improve the behaviour on the trial nodes in the cloud.
```

Comment:
I don't think this just improves over estimates on small jobs (intended the reduction is larger on large jobs). I'd also be inclined to add something about monotonicity. How about:
```suggestion
    // to improve the behaviour on the trial nodes in the cloud. The high level strategy
    // also ensures that corrected memory usage is a monotonic function of estimated
    // memory usage and any change to the approach should preserve this property.
```

Reply:
changed